### ISSUE TITLE
Missing a camera

### ISSUE THREAD

**Issue by mariwing** [twrecked/hass-aarlo#144]
Hi, the Arlo camera counter shows five cameras (which is what I have). But there is only created entities for four of them. Any hints or tips on how to get the fifth camera to appear? Any way to do a rescan to create entities or something like that? Thanks for any advice!

**Comment by twrecked** [comment]
The only way to rescan is to restart the HA daemon - it's a limitation of the HA/Arlo implementation (or, possibly, the way I did it!)

Are the cameras all the same model? On the GUI, if you look under the HAMMER > STATES tab are all the cameras listed there? It will have the name similar to `camera.aarlo_*`

If you can't see it there, are you ok with looking at some debug? If you add this to your `configuration.yaml` and restart HA it will list all the cameras it tried to create in `home-assistant.log`. 

```yaml
logger:
    default: info
    logs:
        pyaarlo: debug
```

You should see some lines like this appear, one per camera:

```
2020-01-07 11:27:47 INFO (MainThread) [custom_components.aarlo.camera] ArloCam: Garage Camera created
```

Just a though, do you have 2 cameras with the same name? I'm not sure how the code would handle that.

**Comment by mariwing** [comment]
Thanks for getting back to me so quickly! Only four of the cameras are listed under states. The cameras are two different models. I have five cameras, four is model VMC4030P and one is model VMC3030. It is one of the VMC4030P that doesn't show up.

They are all named differently. What is a bit interesting though is that some time ago (before trying aarlo component in HASS) I swapped two of the cameras, both the name and location. So lets call them Upstairs and Downstairs. I renamed them so old Upstairs is now called Downstairs and old Downstairs is Upstairs. In Haas Aarlo the entity that are named Downstairs shows the image from the camera previously named Upstairs. The camera that is actually named Upstairs (that was previously called Downstairs) is missing from HAAS! So it could have to do with naming but since the cameras was never used with HAAS/aarlo before the name switch I can't really understand how this affects anything...

In the official Arlo ios app all the cameras show images and are displayed with the correct image/name combination.

Looking at the log it seems that the two cameras both return the same name (even though as explained above one is old name and one is new and in official Arlo app they are NOT named equal

```
2020-01-09 15:19:09 DEBUG (SyncWorker_11) [pyaarlo] set:52M18377AFEE7/deviceId=52M18377AFEE7
2020-01-09 15:19:09 DEBUG (SyncWorker_11) [pyaarlo] set:52M18377AFEE7/deviceName=Loftstua
2020-01-09 15:19:09 DEBUG (SyncWorker_11) [pyaarlo] set:52M1837KB0B71/deviceId=52M1837KB0B71
2020-01-09 15:19:09 DEBUG (SyncWorker_11) [pyaarlo] set:52M1837KB0B71/deviceName=Loftstua

```

**Comment by twrecked** [comment]
2 cameras with the same name will cause a problem with the current system. So:

* I will investigate why the wrong name is being used.
* I'm working on a version of the code that will optionally(!) use the device serial numbers to create entity names, it's in a separate dev branch I just want to test it some more. This will fix your issue.

And:
* Try deleting the `/config/.aarlo/aarlo.pickle` file. Aarlo uses it to store state but it might have got confused.

**Comment by twrecked** [comment]
I just tried changing the camera names on my Arlo installation and everything worked. (Same things happens with code at my job too...)

Did you try logging into my.arlo with your HA Arlo account? The code really should be getting the latest device names and it might be possible to have different names in both accounts.

**Comment by mariwing** [comment]
Interestingly enough; when logging in to the HA Arlo account the cameras had the same names! So I changed it when logged into the my.arlo web and changed the name, restarted HASS and there it was. Everything was correct. So this has to be some kind of bug on Arlo side that made the name change that I did with the "master account" did not propogate to the HA Arlo user?!? 

Anyway, the issue is solved thanks to  your great support so I close it.
